src/hotspot/share/runtime/os.inline.hpp

```diff
@@ -27,6 +27,37 @@
 
 #include "runtime/os.hpp"
 
+#include OS_HEADER(os)
 #include OS_HEADER_INLINE(os)
+#include OS_CPU_HEADER_INLINE(os)
+
+// Below are inline functions that are rarely implemented by the platforms.
+// Provide default emptyy implementation.
+
+#ifndef PLATFORM_PRINT_NATIVE_STACK
+inline bool os::platform_print_native_stack(outputStream* st, const void* context,
+                                     char *buf, int buf_size) {
+  return false;
+}
+#endif
+
+#ifndef HAVE_CDS_CORE_REGION_ALIGNMENT
+inline size_t os::cds_core_region_alignment() {
+  return (size_t)os::vm_allocation_granularity();
+}
+#endif
+
+#ifndef _WINDOWS
+// Currently used only on Windows.
+inline bool os::register_code_area(char *low, char *high) {
+  return true;
+}
+#endif
```

Comment:
We could use some custom define, similar to HAVE_FUNCTION_DESCRIPTORS and remove this default impl just like with `resolve_function_descriptor` below?